model.py

```diff
@@ -1,103 +1,1394 @@
+            inc = roche.findi(q.currVal, dphi.currVal)
+
+            # Calculate wd contact phases 3 and 4
+            phi3, phi4 = roche.wdphases(q.currVal, inc, rwd.currVal, ntheta=10)
+
+            # Calculate length of wd egress
+            dpwd = phi4 - phi3
+
+            # Distance from changepoints to mideclipse
+            dist_cp = (dphi.currVal+dpwd)/2.
+
+            # save these values for speed
+            self._dist_cp = dist_cp
+            self._oldq = q.currVal
+            self._olddphi = dphi.currVal
+            self._oldrwd = rwd.currVal
+        else:
+            # Use the old values
+            dist_cp = self._dist_cp
+
+        # Find location of all changepoints
+        min_ecl = int(np.floor(eclipse.lc.x.min()))
+        max_ecl = int(np.ceil(eclipse.lc.x.max()))
+
+        eclipses = [e for e in range(min_ecl, max_ecl+1)
+                    if np.logical_and(e > eclipse.lc.x.min(),
+                                      e < 1+eclipse.lc.x.max()
+                                      )
+                    ]
+
+        changepoints = []
+        for e in eclipses:
+            # When did the last eclipse end?
+            egress = (e-1) + dist_cp + phi0
+            # When does this eclipse start?
+            ingress = e - dist_cp + phi0
+            changepoints.append([egress, ingress])
+
+        return changepoints
+
+    def create_GP(self, eclipse):
+        """Constructs a kernel, which is used to create Gaussian processes.
+
+        Creates kernels for both inside and out of eclipse,
+        works out the location of any changepoints present, constructs a single
+        (mixed) kernel and uses this kernel to create GPs
+
+        Requires an Eclipse object to create the GP for. """
+
+        # Get objects for ln_ampin_gp, ln_ampout_gp, ln_tau_gp and find the exponential
+        # of their current values
+        ln_ampin = getattr(self, 'ln_ampin_gp')
+        ln_ampout = getattr(self, 'ln_ampout_gp')
+        ln_tau = getattr(self, 'ln_tau_gp')
+
+        ln_ampin_gp = np.exp(ln_ampin.currVal)
+        ln_ampout_gp = np.exp(ln_ampout.currVal)
+        ln_tau_gp = np.exp(ln_tau.currVal)
+
+        # Calculate kernels for both out of and in eclipse WD eclipse
+        # Kernel inside of WD has smaller amplitude than that of outside
+        # eclipse.
+
+        # First, get the changepoints
+        changepoints = self.calcChangepoints(eclipse)
+
+        # We need to make a fairly complex kernel.
+        # Global flicker
+        kernel = ln_ampin_gp * george.kernels.Matern32Kernel(ln_tau_gp)
+        # inter-eclipse flicker
+        for gap in changepoints:
+            kernel += ln_ampout_gp * george.kernels.Matern32Kernel(
+                ln_tau_gp,
+                block=gap
+            )
+
+        # Use that kernel to make a GP object
+        georgeGP = george.GP(kernel, solver=george.HODLRSolver)
+
+        return georgeGP
+
+    def evaluate_model(self):
```

Comment:
Again, there's the evaluate confusion still hanging around here.  What you've written here is ```ln_like```.